projects/storefrontlib/shared/components/ng-select-a11y/ng-select-a11y.directive.ts

```diff
@@ -51,20 +53,47 @@ export class NgSelectA11yDirective implements AfterViewInit {
     observer.observe(this.elementRef.nativeElement, { childList: true });
   }
 
+  /**
+   * When we inside a combo box using JAWS screen reader and press escape key
+   * an escape keyboard event doesn't get fired, instead an AltLeft is fired.
+   */
+  @HostListener('keyup', ['$event'])
+  onKeyUp(event: KeyboardEvent) {
+    if (
+      !this.featureConfigService?.isEnabled('a11yNgSelectCloseDropdownOnEscape')
+    ) {
+      return;
+    }
+    const jawsEscapeCode = 'AltLeft';
+    if (event.code === jawsEscapeCode) {
+      this.selectComponent.close();
+    }
+  }
+
   @Optional() breakpointService = inject(BreakpointService, { optional: true });
 
   @Inject(PLATFORM_ID) protected platformId: Object;
 
   constructor(
     private renderer: Renderer2,
-    private elementRef: ElementRef
+    private elementRef: ElementRef,
+    private selectComponent: NgSelectComponent
```

Comment:
Better to use the `inject(NgSelectComponent` method to add service rather than in our constructor. This is because we have some schematics that require some extra effort to work correctly when we use the constructor method.

Reply:
@Zeyber fixed